agent/utils/compare_versions.go

```diff
@@ -20,6 +20,9 @@ import (
 	"strings"
 )
 
+// zeroPatch is a string to append patch number zero if the major minor version lacks it
+const zeroPatch = ".0"
```

Comment:
Instead of doing this, I think I'd actually prefer implementing a different non-semver version comparator.  API versions from Docker are not semver and making them look like semver can just be a source of confusion.

Reply:
I just wanted to reuse the existing semver comparator due to time constraints for this fix. I agree this is not the right way, but I would prefer to add it to my list, come back and implement a non-semver comparator once this goes out.

Reply:
That's fair.  Want to add a TODO comment and possibly open an issue to track?

Reply:
Sure, will do that.